crates/uv-auth/src/cache.rs

```diff
@@ -93,8 +94,9 @@ impl CredentialsCache {
     /// cached credentials have a username equal to the provided one — otherwise `None` is returned.
     /// If multiple usernames are used per URL, the realm cache should be queried instead.
     pub(crate) fn get_url(&self, url: &Url, username: &Username) -> Option<Arc<Credentials>> {
+        let url = redacted_url(url);
```

Comment:
I think this should all happen in the `trace!` macro, so that it doesn't run at all unless `trace!` is enabled (extremely rare).

Reply:
I can update to use the trace sites. The thinking was that this made it more unlikely a developer would accidentally leak credentials if adding a future trace.

Reply:
Yes, it looks like I was only handling the insert into `CREDENTIALS_CACHE` before. I'm going to add a test for this case since I would have expected it to break.